presto-main/src/main/java/com/facebook/presto/testing/MaterializedRow.java

```diff
@@ -175,14 +163,73 @@ public boolean equals(Object obj)
             if ((obj == null) || (getClass() != obj.getClass())) {
                 return false;
             }
-            ApproximateDouble o = (ApproximateDouble) obj;
-            return Objects.equals(normalizedValue, o.normalizedValue);
+
+            ApproximateNumeric o = (ApproximateNumeric) obj;
+            return Objects.equals(getNormalizedValue(), o.getNormalizedValue());
         }
 
         @Override
         public int hashCode()
         {
-            return Objects.hash(normalizedValue);
+            return Objects.hash(getNormalizedValue());
+        }
+    }
+
+    private static class ApproximateDouble
+            extends ApproximateNumeric
+    {
+        private final Number value;
```

Comment:
I overlooked it after change requested by @raghavsethi. Here's the fixup: 52b658e35986e55fc4f17090925896ba68be3178